src/shared/data-directory/src/main/java/org/geonetwork/metadatastore/MetadataResource.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * (c) 2003 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license,
+ * available at the root application directory.
+ */
+package org.geonetwork.metadatastore;
+
+import java.util.Date;
+
+public interface MetadataResource {
+
+    String getId();
+
+    String getUrl();
+
+    MetadataResourceVisibility getVisibility();
+
+    long getSize();
+
+    Date getLastModification();
```

Comment:
It would be nice if all api's in GN5 would use iso dates.  In GN4, this date is shown as a unix integer and it causes issues when using codegen and is also not human readable.

I have been wanting to change this in GN4 however I don't know how to change it without breaking backward compatibility.